language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/executors/CreateFunctionExecutor.java

```diff
@@ -89,153 +97,99 @@ public Object execute(ExecuteCommandContext context) throws LSCommandExecutorExc
-//        SyntaxTree syntaxTree;
-//        try {
-//            syntaxTree = docManager.getTree(path);
-//        } catch (WorkspaceDocumentException e) {
-//            throw new LSCommandExecutorException("Error while retrieving syntax tree for path: " + path.toString());
-//        }
-
-//        BLangInvocation functionNode = null;
-//        Position pos = new Position(line, column + 1);
-//        PositionDetails cursorDetails = CodeActionUtil.findCursorDetails(new Range(pos, pos), syntaxTree, context);
-
-//        if (functionNode == null) {
-//            throw new LSCommandExecutorException("Couldn't find the function node!");
-//        }
-//        String functionName = functionNode.name.getValue();
-
-//        BLangNode parent = functionNode.parent;
-//        BLangPackage packageNode = CommonUtil.getPackageNode(functionNode);
-
-        int eLine = 0;
-        int eCol = 0;
-
-//        BLangCompilationUnit cUnit = getCurrentCUnit(context, packageNode);
-//        for (TopLevelNode topLevelNode : cUnit.topLevelNodes) {
-//            if (topLevelNode.getPosition().lineRange().endLine().line() > eLine) {
-//                eLine = topLevelNode.getPosition().lineRange().endLine().line();
-//            }
-//        }
 
-        List<TextEdit> edits = new ArrayList<>();
-//        if (parent != null && packageNode != null) {
-//            PackageID currentPkgId = packageNode.packageID;
-//            ImportsAcceptor importsAcceptor = new ImportsAcceptor(context);
-        //TODO: Fix this when #26382 is avaialble
-
-//            returnType = FunctionGenerator.generateTypeDefinition(importsAcceptor, currentPkgId, parent, context);
-//            returnValue = FunctionGenerator.generateReturnValue(importsAcceptor, currentPkgId, parent,
-//                                                                "    return {%1};", context);
-//            List<String> arguments = FunctionGenerator.getFuncArguments(importsAcceptor, currentPkgId, functionNode,
-//                                                                        context);
-//            if (arguments != null) {
-//                funcArgs = String.join(", ", arguments);
-//            }
-//            edits.addAll(importsAcceptor.getNewImportTextEdits());
-//        } else {
-//            throw new LSCommandExecutorException("Error occurred when retrieving function node!");
-//        }
+        LineRange rootLineRange = syntaxTree.rootNode().lineRange();
+        int endLine = rootLineRange.endLine().line() + 1;
+        int endCol = 0;
+        // If the file ends with a new line
+        boolean newLineAtEnd = rootLineRange.endLine().offset() == 0;
+
+        Document document = context.workspace().document(filePath.get()).orElseThrow();
+        LinePosition linePosition = LinePosition.from(endLine, endCol);
+        Set<String> visibleSymbolNames = semanticModel.visibleSymbols(document, linePosition).stream()
```

Comment:
Seems like not. Because we get the `ExecuteCommandContext` here. Not `CodeActionContext`.

Reply:
Then, shall we add the API to the `ExecuteCommandContext` as well? WDYT?

Reply:
The `CodeActionContext` is extended from `DocumentServiceContext` while `ExecuteCommandContext` is extended from `WorkspaceServiceContext` (which isn't related to a specific file). Therefore, if we are to add the mentioned method, it should be like this:
```
    List<Symbol> visibleSymbols(Path filePath, Position position);
```
Will that be fine?

Reply:
Yeah, that would be fine. Id we are gonna invoke the visible symbols, API multiple times within a given operation flow, let's add the API. Otherwise let's use your approach for the moment

Reply:
Added it to the `WorkspacServiceContext`.